packages/strapi-design-system/src/Select/Select.js

```diff
@@ -0,0 +1,196 @@
+
+    const handleChange = () => {
+      onChange(node.props.value);
+      setExpanded(false);
+      buttonRef.current.focus();
+    };
+
+    const selected = node.props.value === value;
+
+    if (selected) {
+      selectOptionLabel = node.props.children;
+      activeOptionId = optionId;
+    }
+
+    return cloneElement(node, {
+      id: optionId,
+      onSelect: handleChange,
+      selected,
+    });
+  });
+
+  return (
+    <Field hint={hint} error={error} id={idRef.current}>
+      <Stack size={1}>
+        <FieldLabel as="span" id={labelId}>
+          {label}
+        </FieldLabel>
+
+        <SelectButtonWrapper hasError={Boolean(error)} disabled={disabled} ref={containerRef}>
+          <Row justifyContent="space-between" as="span">
+            <SelectButton
+              ref={buttonRef}
+              labelledBy={selectOptionLabel ? `${labelId} ${contentId}` : labelId}
+              expanded={expanded}
+              onTrigger={handleTrigger}
+              id={idRef.current}
+              hasError={Boolean(error)}
+              disabled={disabled}
+              onMouseDown={handleMouseDown}
+              {...props}
+            >
+              <Text id={contentId} as="span" aria-hidden={true} textColor={value ? 'neutral800' : 'neutral600'}>
+                {selectOptionLabel || placeholder}
+              </Text>
+            </SelectButton>
+
+            <Row>
+              {value && (
+                <IconBox as="button" onClick={handleClear} aria-label={clearLabel} aria-disabled={disabled}>
+                  <CloseAlertIcon />
+                </IconBox>
+              )}
+
+              <CaretBox paddingLeft={3} aria-hidden as="button" onMouseDown={handleMouseDown} tabIndex={-1}>
+                <DropdownIcon />
+              </CaretBox>
+            </Row>
+          </Row>
+        </SelectButtonWrapper>
+
+        <FieldHint />
+        <FieldError />
+      </Stack>
+
+      {expanded && (
+        <Popover source={containerRef} spacingTop={1} fullWidth>
+          <SelectList
+            selectId={idRef.current}
+            selectedOptionId={activeOptionId}
+            labelledBy={labelId}
+            ref={listRef}
+            onEscape={handleEscape}
+          >
+            {childrenClone}
+          </SelectList>
+        </Popover>
+      )}
+    </Field>
+  );
+};
+
+Select.defaultProps = {
+  children: [],
+  disabled: false,
+  id: undefined,
+  placeholder: undefined,
+  value: undefined,
+  hint: undefined,
+  error: undefined,
+};
+
+Select.propTypes = {
+  children: PropTypes.arrayOf(PropTypes.node),
+  clearLabel: PropTypes.string.isRequired,
+  disabled: PropTypes.bool,
+  error: PropTypes.string,
+  hint: PropTypes.string,
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  label: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
+  placeholder: PropTypes.string,
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
I think that the value should allow more types like `object`

Reply:
It's a string or a number right now (in the "option" component)

Reply:
Do you think that we could also pass other types? 
We often do this and I think the select should support it.
WDYT?

Reply:
Value is the thing you pass to regular "option". You can't pass objects https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select